basicdataset/src/main/java/ai/djl/basicdataset/nlp/StanfordQuestionAnsweringDataset.java

```diff
@@ -228,6 +228,30 @@ public Object getData() throws IOException {
         return data;
     }
 
+    /**
+     * Since a question might have no answer, we need extra logic to find the last index of the
+     * answer in the {@code TargetTextData}. There are not many consecutive questions without
+     * answer, so this logic will not cause a high cost.
+     *
+     * @param questionInfoIndex the last index of the record in {@code questionInfoList} that needs
+     *     to be preprocessed
+     * @return the last index of the answer in {@code TargetTextData} that needs to be preprocessed
+     */
+    private int getLastAnswerIndex(int questionInfoIndex) {
+        boolean endAtHead = false;
+        QuestionInfo questionInfo = questionInfoList.get(questionInfoIndex);
+        while (questionInfo.answerIndexList.isEmpty()) {
+            if (questionInfoIndex == 0) {
+                endAtHead = true;
+                break;
+            }
+            questionInfo = questionInfoList.get(--questionInfoIndex);
+        }
+        return endAtHead
+                ? 0
+                : questionInfo.answerIndexList.get(questionInfo.answerIndexList.size() - 1);
+    }
```

Comment:
```suggestion
    private int getLastAnswerIndex(int questionInfoIndex) {
        // Go backwards through the questionInfoList until it finds one with an answer
        for (; questionInfoIndex >= 0; questionInfoIndex--) {
            QuestionInfo questionInfo = questionInfoList.get(questionInfoIndex);
            if (!questionInfo.answerIndexList.isEmpty()) {
                return questionInfo.answerIndexList.get(questionInfo.answerIndexList.size() - 1);
            }
        }

        // Could not find a QuestionInfo with an answer
        return 0;
    }
```

I found this function a little difficult to understand, so I wanted to suggest a rewrite that might seem a bit clearer. One thing to pay attention to when writing code which is easy to read is to match what people expect. When we write loops that go through a list, it is almost always a for loop. So, that can fit expectations a bit more than using a while loop, even if the backwards for loop with an existing variable is a bit less common type of for loop. It also simplifies control flow a bit using a `return` to exit from your loop early rather than a `break` and then a conditional after the `break`.